emu/src/cpu/arm7tdmi.rs

```diff
@@ -903,6 +903,26 @@ impl Arm7tdmi {
 
         Some(SIZE_OF_THUMB_INSTRUCTION)
     }
+
+    fn long_branch_link(&mut self, op_code: ThumbModeOpcode) -> Option<u32> {
+        let h = op_code.get_bit(11);
+        let offset = op_code.get_bits(0..=10);
+
+        if h {
+            let next_instruction =
+                self.registers.program_counter() as u32 + SIZE_OF_THUMB_INSTRUCTION;
+            let lr = self.registers.register_at(REG_LR);
+            let offset = (offset << 1) as u32;
+            self.registers.set_program_counter(lr + offset);
+            self.registers.set_register_at(REG_LR, next_instruction | 1);
```

Comment:
Did you understand why we set bit 0? I can't figure it out why

Reply:
Is it because if we later do a `bx lr` then we take the bit 0 as `1` and it means we would still use THUMB?

Reply:
Don't remember right now, I need to recall my walkthrough ehehe but maybe yes!